src/hera/workflows/_meta_mixins.py

```diff
@@ -372,6 +372,12 @@ def __call__(self, *args, **kwargs) -> Union[None, Step, Task]:
                 return Step(template=self, **kwargs)
 
             if isinstance(_context.pieces[-1], DAG):
+                # Add dependencies based on context if not explicitly provided
+                current_task_depends = _context.pieces[-1]._current_task_depends
+                if current_task_depends and "depends" not in kwargs:
+                    kwargs["depends"] = " && ".join(sorted(current_task_depends))
+                current_task_depends.clear()
```

Comment:
Pre-existing issue: if the user creates a Task explicitly rather than via a `@script`-decorated function, the dependencies list won't be cleared. Perhaps this should be moved to the point where a Task is registered?

Reply:
Yes, makes sense to move it to DAG's `_add_sub` https://github.com/argoproj-labs/hera/blob/e7f38fa958a70e3f8cdf66471263f2656511f7ff/src/hera/workflows/dag.py#L50-L56

Reply:
Done.